### PR TITLE
eliminate LoadUser via loadMe in teams

### PR DESCRIPTION
**Pull request by maxtaco**
- we can use the fullSelfCacher with a few tweaks, as seen here

### CODE REVIEW

File: go/teams/teams.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ type Team struct {
 
 	keyManager *TeamKeyManager
 
-	me      *libkb.User
-	rotated bool
+	meForSignature libkb.UserForSignatures
```
Comment (mlsteele) [inline go/teams/teams.go]: This field still starts out unset. Make it a pointer. Having `nil` is less likely to be accidentally used than a blank `libkb.UserForSignatures`

Comment (mlsteele) [reply go/teams/teams.go]: Or maybe it's not used at all?

File: go/teams/sig.go [hunk 2]
```diff
@@ -18,15 +18,16 @@ import (
...
+func TeamRootSig(g *libkb.GlobalContext, me libkb.UserForSignatures, key libkb.GenericKey, teamSection SCTeamSection) (*jsonw.Wrapper, error) {
 	ret, err := libkb.ProofMetadata{
-		Me:         me,
-		LinkType:   libkb.LinkTypeTeamRoot,
-		SigningKey: key,
-		Seqno:      1,
-		SigVersion: libkb.KeybaseSignatureV2,
-		SeqType:    seqTypeForTeamPublicness(teamSection.Public),
-	}.ToJSON(me.G())
+		SigningUser: me,
```
Comment (mlsteele) [inline go/teams/sig.go]: That's a pretty nice use of interfaces

Comment (maxtaco) [reply go/teams/sig.go]: gee thanks!

**Pull request merged**